app/main.py

```diff
@@ -1,2 +1,106 @@
+from typing import Any, Hashable, Optional, List, Iterable, Union, Tuple
+
+
 class Dictionary:
-    pass
+    class _Node:
+        def __init__(self, key: Hashable, value: Any) -> None:
+            self.key = key
+            self.hash_key = hash(key)
+            self.value = value
+
+    def __init__(
+            self,
+            initial_capacity: int = 8,
+            load_factor: float = 2 / 3
+    ) -> None:
+        self._capacity: int = initial_capacity
+        self._load_factor: float = load_factor
+        self._size: int = 0
+        self._buckets: (
+            List)[Optional[Dictionary._Node]] = [None] * self._capacity
+
+    def _get_index(self, key: Hashable) -> int:
+        return hash(key) % self._capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        if self._size / self._capacity >= self._load_factor:
+            self._resize()
+        index = self._get_index(key)
+        while self._buckets[index] is not None:
```

Comment:
this while loop repeats in all your get/set/delete methods with slight variations. Maybe place it in separate method and reuse it throughout your code?